### ISSUE TITLE
Can I Show preview screen when taking photos by python?

### ISSUE THREAD

**Issue by Skullface9512** [TheImagingSource/ic4-examples#19]
Hi, in python tutorials, i know that how to set some relative properties and grab images.  To know that what the environment i'm grabbing, i always open the IC Capture 2.5 and it's very troublesome, can i get image data from buffer directly using python and show it by opencv

**Comment by TIS-Tim** [comment]
If you want to display an image buffer using OpenCV, you can call numpy_wrap (or numpy_copy, if the buffer's pixel format is not directly supported) on it and pass the result to imshow as shown [here](https://github.com/TheImagingSource/ic4-examples/blob/master/python/thirdparty-integration/imagebuffer-numpy-opencv-snap/imagebuffer-numpy-opencv-snap.py).

You can also create an ic4 display and manually select which buffer should be displayed on it as shown in [this example](https://github.com/TheImagingSource/ic4-examples/blob/master/python/thirdparty-integration/imagebuffer-numpy-opencv-live/imagebuffer-numpy-opencv-live.py).

Finally, if you happen to use Qt (PySide6) you can embed the display into your application window as shown [here](https://github.com/TheImagingSource/ic4-examples/blob/master/python/qt6/qt6-first-steps/qt6-first-steps.py).

(The latter example should really be using the (new in 1.1) imagingcontrol4.pyside6.DisplayWidget, it needs an update.)

**Comment by Skullface9512** [comment]
Hi, I have another problem, I need to adjust the camera's focal length through python script, do you provide some relative interface in python

**Comment by TIS-Tim** [comment]
For focal length adjustment via software you do need a zoom camera.

If you have one of those, it will be as simple as this:

```
grabber.device_property_map.set_value(imagingcontrol4.PropId.ZOOM, 123)
```

Here 123 is the logical zoom value. If you have a 39G camera, to get an approximate focal length of that number, have a look at the camera's technical reference manual.

**Comment by Skullface9512** [comment]
The camera i used is DFM 37UX178-ML, can i adjust its focal length through the method you describe above.

**Comment by TIS-Tim** [comment]
No. Focal length is a hardware feature of the lens. If you want to adjust focal length via software, you need a motorized lens.
